src/compressed_chunk.c

```diff
@@ -217,11 +218,7 @@ ChunkResult Compressed_ChunkIteratorGetNext(ChunkIter_t *iter, Sample *sample) {
     return Compressed_ReadNext((Compressed_Iterator *)iter, &sample->timestamp, &sample->value);
 }
 
-void Compressed_FreeChunkIterator(ChunkIter_t *iter, bool rev) {
-    // compressed iterator on reverse query has to release decompressed chunk
-    if (rev) {
-        free(((ChunkIterator *)iter)->chunk);
-    }
+void Compressed_FreeChunkIterator(ChunkIter_t *iter) {
```

Comment:
@danni-m don't you need to free the chunk also in the compressed case?

```
    if (iter->options & CHUNK_ITER_OP_FREE_CHUNK) {
        free(iter->chunk);
    }
```

Reply:
see comment above.